packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -429,6 +501,131 @@ describe("Presence", () => {
+
+							// Act - disconnect & reconnect local client
+							runtime.disconnect(); // Simulate local client disconnect
+							clock.tick(1000);
+							runtime.connect("client6"); // Simulate local client reconnect with new connection id
+
+							// Verify - stale attendee should still be 'Connected' after 15 seconds
+							clock.tick(15_001);
+							assert.strictEqual(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Connected,
+								"Stale attendee should still be 'Connected' after 15s",
+							);
+
+							// Verify - stale attendee should be 'Disconnected' after 30 seconds and announced via `attendeeDisconnected`
+							clock.tick(15_001);
+							assert.strictEqual(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Disconnected,
+								"Stale attendee should be 'Disconnected' 30s after reconnection",
+							);
+							assert.strictEqual(
+								disconnectedAttendees.length,
+								1,
+								"Exactly one attendee should be announced as disconnected",
+							);
+						});
+
+						it.skip("does not update stale attendee status if local client does not reconnect", () => {
+							assert(knownAttendee !== undefined, "No attendee was set in beforeEach");
+
+							// Act - disconnect local client
+							runtime.disconnect();
+
+							// Verify - stale attendee should still be 'Connected' after 30 seconds
+							clock.tick(30_001);
+							assert.strictEqual(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Connected,
+								"Stale attendee should still be 'Connected' after 30s",
+							);
+						});
+
+						it.skip("does not update stale attendee status if local client reconnection lasts less than 30s", () => {
+							assert(knownAttendee !== undefined, "No attendee was set in beforeEach");
+
+							// Act - disconnect, reconnect for 15 second, then disconnect local client again
+							runtime.disconnect(); // First disconnect
+							clock.tick(1000);
+							runtime.connect("client6"); // Reconnect
+							clock.tick(15_001); // Wait 15 seconds
+							runtime.disconnect(); // Disconnect again
+							clock.tick(15_001); // Wait another 15 seconds (30s total since reconnection)
+
+							// Verify - stale attendee should still be 'Connected' after 30s post-reconnection
+							assert.strictEqual(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Connected,
+								"Stale attendee should still be 'Connected' after 30s",
+							);
+						});
+
+						it.skip("does not update attendee status to 'Disconnected' if stale attendee becomes active", () => {
+							assert(knownAttendee !== undefined, "No attendee was set in beforeEach");
+
+							// Act - disconnect, reconnect, then process rejoin signal from known attendee after 15s
+							runtime.disconnect();
+							clock.tick(1000);
+							runtime.connect("client6");
+							clock.tick(15_001);
+							processJoinSignals([rejoinAttendeeSignal]);
```

Comment:
How else might a stale attendee become active. This sounds like situation where the other was also disconnected but has rejoined. What about if the other had always stayed connected?

Reply:
Added a case for processing a datastore update message. This should also activate stale client